android/app/src/main/java/com/zulipmobile/notifications/GCMPushNotifications.java

```diff
@@ -37,14 +37,14 @@
 
     public static final int NOTIFICATION_ID = 435;
     public static final String ACTION_NOTIFICATIONS_DISMISS = "ACTION_NOTIFICATIONS_DISMISS";
-    private LinkedHashMap<String, Pair<String, Integer>> conversations;
+    private LinkedHashMap<String, List<MessageInfo>> conversations;
```

Comment:
I'd really like a javadoc comment on this saying
* what these keys are
* that the lists have the latest message first.

That last bit, in particular, is a (useful!) bit of complexity that is added here, so this is a good time to start writing stuff down.

Reply:
Thanks for adding a comment! It didn't actually quite answer either of my two questions, though. :-)

I added a commit on top, cf1cd2a1a, to add those details; take a look to see what I had in mind.